src/main/java/org/testng/reporters/JUnitReportReporter.java

```diff
@@ -214,34 +214,29 @@ private TestTag createTestTagFor(ITestResult tr, Class<?> cls) {
     Properties p2 = new Properties();
     p2.setProperty(XMLConstants.ATTR_CLASSNAME, cls.getName());
     p2.setProperty(XMLConstants.ATTR_NAME, getTestName(tr));
-    Throwable t = tr.getThrowable();
-
-    switch (tr.getStatus()) {
-      case ITestResult.SKIP:
-      case ITestResult.SUCCESS_PERCENTAGE_FAILURE:
-        testTag.childTag = XMLConstants.SKIPPED;
-        break;
-
-      case ITestResult.FAILURE:
-        testTag.childTag = XMLConstants.ERROR;
-        if (t instanceof AssertionError) {
-          testTag.childTag = XMLConstants.FAILURE;
-        }
-        if (t != null) {
-          StringWriter sw = new StringWriter();
-          PrintWriter pw = new PrintWriter(sw);
-          t.printStackTrace(pw);
-          testTag.message = t.getMessage();
-          testTag.type = t.getClass().getName();
-          testTag.stackTrace = sw.toString();
-        }
-        break;
+    int status = tr.getStatus();
+    if (status == ITestResult.SKIP || status == ITestResult.SUCCESS_PERCENTAGE_FAILURE) {
+      testTag.childTag = XMLConstants.SKIPPED;
+    } else if (status == ITestResult.FAILURE) {
+      handleFailure(testTag, tr.getThrowable());
     }
-
     testTag.properties = p2;
     return testTag;
   }
 
+  private static void handleFailure(TestTag testTag, Throwable t) {
+    testTag.childTag = XMLConstants.ERROR;
+    if (t instanceof AssertionError) {
```

Comment:
Note to self: The bug is here. While refactoring, the changes causes the exception details to be packed only for assertion errors but for errors (tests failing due to unexpected exceptions), the exception details are not captured.